identity/v3userpass.go

```diff
@@ -103,13 +103,14 @@ func (u *V3UserPass) Auth(creds *Credentials) (*AuthDetails, error) {
 			},
 		},
 	}
-	if creds.TenantName != "" {
+	if creds.TenantName != "" || creds.TenantID != "" {
```

Comment:
This will break things. Having this means that we are totally ignoring project domain.